hi-ml-histopathology/src/histopathology/datamodules/base_module.py

```diff
@@ -302,11 +307,15 @@ def _load_dataset(self, slides_dataset: SlidesDataset) -> Dataset:
                 ),
             ]
         )
-        transforms = Compose([base_transform, self.transform]).flatten() if self.transform else base_transform
+        if self.transform and self.transform[stage]:
+            transforms = Compose([base_transform, self.transform[stage]]).flatten()
+        else:
+            transforms = base_transform
         return Dataset(slides_dataset, transforms)
 
-    def _get_dataloader(self, dataset: SlidesDataset, shuffle: bool, **dataloader_kwargs: Any) -> DataLoader:
-        transformed_slides_dataset = self._load_dataset(dataset)
+    def _get_dataloader(self, dataset: SlidesDataset, stage: str, shuffle: bool,
```

Comment:
should stage be a string? On lines 225, 228, 231 when we call this method, I think stage will have 'ModelKey' enum type, unless we call `.value` on it

Reply:
All enums in `naming.py` extend both `str` and `Enum` to allow interoperability.